chips/stm32f4xx/src/rcc.rs

```diff
@@ -733,6 +734,7 @@ pub(crate) const DEFAULT_PLLQ_VALUE: PLLQ = match DEFAULT_PLLM_VALUE {
 
 pub struct Rcc {
     registers: StaticRef<RccRegisters>,
+    hse_frequency: OptionalCell<usize>,
```

Comment:
Is there any reason why `hse_frequency` does not belong to `Hse` struct? I would keep `Rcc` as a wrapper over register manipulation and move all high-level logic to corresponding structs, such as `Hsi`, `Hse` and `Pll`.

Reply:
I would add `hse` as a member of `pll`.

Reply:
Done. `hse` is now a public member of `pll`. Any board that would like to use HSE will have to set the HSE frequency using `pll`.

Reply:
So how will `pll` retrieve the HSE frequency? `pll` does not have any interface to `clocks`, only to `rcc`.

Reply:
Both `Clocks` and `Pll` will have `Hse` as a member. In the case of `Clocks`, it will be a public member, and in the case of `Pll`, a private member. `Pll` will use `Hse` to compute N, P, Q parameters when `set_frequency()` is called. The public member of `Clocks` will be used inside the board file to configure it depending on the real hardware configuration. Keeping `Hse` as a member of `Clocks` doesn't exclude the possibility of adding it to `Pll`.

The downside of this approach is storing two references to the same object. The upside, in my opinion, is clean code structure that reflects the real world.

What do you think?

Reply:
I implemented something a bit different. `hse` is a member of `clocks` only.
Configurating the PLL frequency is done from `clocks` specifying the PLL source and its frequency. for example:
`clocks.set_pll_frequency(stm32f429zi::rcc::PllSource::HSE, 96);`
`clocks` will retrieve the HSE frequency if needed.
Let me know what do you think

Reply:
Your solution works as well. The sole question that I have: would anyone ever want to call `Pll::set_frequency` directly? If not, I would suggest making it `pub(in super)` to restrict its visibility to `clocks` module only.